friends/src/reducers/friendsReducer.js

```diff
@@ -0,0 +1,47 @@
+import {
+    FETCHING_FRIENDS,
+    FETCHING_FRIENDS_SUCCESS,
+    FETCHING_FRIENDS_FAILURE,
+    CREATING_FRIEND,
+    CREATING_FRIEND_SUCCESS,
+    CREATING_FRIEND_FAILURE
+  } from '../actions';
+  
+  const initialState = {
+    friends: [],
+  
+    // get friends
+    fetchingFriends: false,
+  
+    // create new friend
+    creatingFriend: false,
+  
+    error: null
+  }
+  
+  export const friendsReducer = (state = initialState, action) => {
+    switch (action.type) {
+      case FETCHING_FRIENDS:
+        return { ...state, fetchingFriends: true };
+      case FETCHING_FRIENDS_SUCCESS:
+        return {
+          ...state,
+          fetchingFriends: false,
+          friends: [...state.friends, ...action.payload]
+        };
+      case FETCHING_FRIENDS_FAILURE:
+        return { ...state, fetchingFriends: false, error: action.payload };
+      case CREATING_FRIEND:
+        return { ...state, creatingFriend: true };
+      case CREATING_FRIEND_SUCCESS:
+        return {
+          ...state,
+          creatingFriend: false,
+          friends: [...state.friends, ...action.payload]
```

Comment:
Because of the way You destructure the state Your `getFriends` method in `FriendListContainer` will return duplicate entries after adding new friend. After refresh everything will look alright but this is an issue that needs fixing.

Simply doing the following will fix the issue and not return duplicate entries anymore.
```
        return {
          ...state,
          creatingFriend: false,
          friends: [...action.payload]
}
```